REST restrictions endpoints returns 500 instead of 403 if user has no rights

Other newer endpoints might be affected as well.

Exception:
```
2018-04-10 17:08:31,142 logLev=ERROR class=org.apache.catalina.core.ContainerBase.[jboss.web].[default-host].[/AMW_rest].[ch.mobi.itc.mobiliar.rest.RESTApplication] thread="ajp-/0.0.0.0:8009-6" JBWEB000236: Servlet.service() for servlet ch.mobi.itc.mobiliar.rest.RESTApplication threw exception: org.jboss.resteasy.spi.UnhandledException: javax.ejb.EJBTransactionRolledbackException: Not Authorized!
        at org.jboss.resteasy.core.SynchronousDispatcher.handleApplicationException(SynchronousDispatcher.java:369)
        at org.jboss.resteasy.core.SynchronousDispatcher.handleException(SynchronousDispatcher.java:237)
        at org.jboss.resteasy.core.SynchronousDispatcher.handleInvokerException(SynchronousDispatcher.java:213)
        at org.jboss.resteasy.core.SynchronousDispatcher.getResponse(SynchronousDispatcher.java:598)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:565)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:130)
        at org.jboss.resteasy.plugins.server.servlet.ServletContainerDispatcher.service(ServletContainerDispatcher.java:208)
        at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:55)
        at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:50)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:847)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:295)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:214)
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:231)
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:149)
        at org.jboss.modcluster.container.jbossweb.JBossWebContext$RequestListenerValve.event(JBossWebContext.java:91)
        at org.jboss.modcluster.container.jbossweb.JBossWebContext$RequestListenerValve.invoke(JBossWebContext.java:72)
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:512)
        at org.jboss.as.jpa.interceptor.WebNonTxEmCloserValve.invoke(WebNonTxEmCloserValve.java:50)
        at org.jboss.as.jpa.interceptor.WebNonTxEmCloserValve.invoke(WebNonTxEmCloserValve.java:50)
        at org.jboss.as.web.security.SecurityContextAssociationValve.invoke(SecurityContextAssociationValve.java:169)
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:151)
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:97)
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:102)
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
        at org.apache.coyote.ajp.AjpProcessor.process(AjpProcessor.java:490)
        at org.apache.coyote.ajp.AjpProtocol$AjpConnectionHandler.process(AjpProtocol.java:422)
        at org.apache.tomcat.util.net.JIoEndpoint$Worker.run(JIoEndpoint.java:926)
        at java.lang.Thread.run(Thread.java:748) [rt.jar:1.8.0_144]
Caused by: javax.ejb.EJBTransactionRolledbackException: Not Authorized!
        at org.jboss.as.ejb3.tx.CMTTxInterceptor.handleInCallerTx(CMTTxInterceptor.java:161)
        at org.jboss.as.ejb3.tx.CMTTxInterceptor.invokeInCallerTx(CMTTxInterceptor.java:260)
        at org.jboss.as.ejb3.tx.CMTTxInterceptor.required(CMTTxInterceptor.java:333)
        at org.jboss.as.ejb3.tx.CMTTxInterceptor.processInvocation(CMTTxInterceptor.java:243)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.component.interceptors.CurrentInvocationContextInterceptor.processInvocation(CurrentInvocationContextInterceptor.java:41)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.component.invocationmetrics.WaitTimeInterceptor.processInvocation(WaitTimeInterceptor.java:43)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.deployment.processors.StartupAwaitInterceptor.processInvocation(StartupAwaitInterceptor.java:22)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.component.interceptors.ShutDownInterceptorFactory$1.processInvocation(ShutDownInterceptorFactory.java:64)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.component.interceptors.LoggingInterceptor.processInvocation(LoggingInterceptor.java:59)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ee.component.NamespaceContextInterceptor.processInvocation(NamespaceContextInterceptor.java:50)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.component.interceptors.AdditionalSetupInterceptor.processInvocation(AdditionalSetupInterceptor.java:55)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ee.component.TCCLInterceptor.processInvocation(TCCLInterceptor.java:45)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.invocation.ChainedInterceptor.processInvocation(ChainedInterceptor.java:61)
        at org.jboss.as.ee.component.ViewService$View.invoke(ViewService.java:189)
        at org.jboss.as.ee.component.ViewDescription$1.processInvocation(ViewDescription.java:185)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.invocation.ChainedInterceptor.processInvocation(ChainedInterceptor.java:61)
        at org.jboss.as.ee.component.ProxyInvocationHandler.invoke(ProxyInvocationHandler.java:73)
        at ch.puzzle.itc.mobiliar.business.security.control.PermissionService$$$view56.throwNotAuthorizedException(Unknown Source)
        at sun.reflect.GeneratedMethodAccessor578.invoke(Unknown Source) [:1.8.0_144]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) [rt.jar:1.8.0_144]
        at java.lang.reflect.Method.invoke(Method.java:498) [rt.jar:1.8.0_144]
        at org.jboss.weld.util.reflection.SecureReflections$13.work(SecureReflections.java:267)
        at org.jboss.weld.util.reflection.SecureReflectionAccess.run(SecureReflectionAccess.java:52)
        at org.jboss.weld.util.reflection.SecureReflectionAccess.runAsInvocation(SecureReflectionAccess.java:137)
        at org.jboss.weld.util.reflection.SecureReflections.invoke(SecureReflections.java:263)
        at org.jboss.weld.bean.proxy.EnterpriseBeanProxyMethodHandler.invoke(EnterpriseBeanProxyMethodHandler.java:115)
        at org.jboss.weld.bean.proxy.EnterpriseTargetBeanInstance.invoke(EnterpriseTargetBeanInstance.java:56)
        at org.jboss.weld.bean.proxy.ProxyMethodHandler.invoke(ProxyMethodHandler.java:105)
        at ch.puzzle.itc.mobiliar.business.security.control.PermissionService$Proxy$_$$_Weld$Proxy$.throwNotAuthorizedException(PermissionService$Proxy$_$$_Weld$Proxy$.java)
        at ch.puzzle.itc.mobiliar.business.security.interceptor.HasPermissionInterceptor.roleCall(HasPermissionInterceptor.java:124)
        at sun.reflect.GeneratedMethodAccessor356.invoke(Unknown Source) [:1.8.0_144]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) [rt.jar:1.8.0_144]
        at java.lang.reflect.Method.invoke(Method.java:498) [rt.jar:1.8.0_144]
        at org.jboss.as.ee.component.ManagedReferenceLifecycleMethodInterceptor.processInvocation(ManagedReferenceLifecycleMethodInterceptor.java:89)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.invocation.WeavedInterceptor.processInvocation(WeavedInterceptor.java:53)
        at org.jboss.as.ee.component.interceptors.UserInterceptorFactory$1.processInvocation(UserInterceptorFactory.java:63)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.component.invocationmetrics.ExecutionTimeInterceptor.processInvocation(ExecutionTimeInterceptor.java:43)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.jpa.interceptor.SBInvocationInterceptor.processInvocation(SBInvocationInterceptor.java:47)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.weld.ejb.EjbRequestScopeActivationInterceptor.processInvocation(EjbRequestScopeActivationInterceptor.java:73)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.invocation.InitialInterceptor.processInvocation(InitialInterceptor.java:21)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.invocation.ChainedInterceptor.processInvocation(ChainedInterceptor.java:61)
        at org.jboss.as.ee.component.interceptors.ComponentDispatcherInterceptor.processInvocation(ComponentDispatcherInterceptor.java:53)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.component.pool.PooledInstanceInterceptor.processInvocation(PooledInstanceInterceptor.java:51)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.tx.CMTTxInterceptor.invokeInOurTx(CMTTxInterceptor.java:279)
        at org.jboss.as.ejb3.tx.CMTTxInterceptor.requiresNew(CMTTxInterceptor.java:348)
        at org.jboss.as.ejb3.tx.CMTTxInterceptor.processInvocation(CMTTxInterceptor.java:245)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.component.interceptors.CurrentInvocationContextInterceptor.processInvocation(CurrentInvocationContextInterceptor.java:41)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.component.invocationmetrics.WaitTimeInterceptor.processInvocation(WaitTimeInterceptor.java:43)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.deployment.processors.StartupAwaitInterceptor.processInvocation(StartupAwaitInterceptor.java:22)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.component.interceptors.ShutDownInterceptorFactory$1.processInvocation(ShutDownInterceptorFactory.java:64)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.component.interceptors.LoggingInterceptor.processInvocation(LoggingInterceptor.java:59)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ee.component.NamespaceContextInterceptor.processInvocation(NamespaceContextInterceptor.java:50)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.component.interceptors.AdditionalSetupInterceptor.processInvocation(AdditionalSetupInterceptor.java:55)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ee.component.TCCLInterceptor.processInvocation(TCCLInterceptor.java:45)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.invocation.ChainedInterceptor.processInvocation(ChainedInterceptor.java:61)
        at org.jboss.as.ee.component.ViewService$View.invoke(ViewService.java:189)
        at org.jboss.as.ee.component.ViewDescription$1.processInvocation(ViewDescription.java:185)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.invocation.ChainedInterceptor.processInvocation(ChainedInterceptor.java:61)
        at org.jboss.as.ee.component.ProxyInvocationHandler.invoke(ProxyInvocationHandler.java:73)
        at ch.puzzle.itc.mobiliar.business.security.boundary.PermissionBoundary$$$view59.getAllAvailablePermissions(Unknown Source)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) [rt.jar:1.8.0_144]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) [rt.jar:1.8.0_144]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) [rt.jar:1.8.0_144]
        at java.lang.reflect.Method.invoke(Method.java:498) [rt.jar:1.8.0_144]
        at org.jboss.weld.util.reflection.SecureReflections$13.work(SecureReflections.java:267)
        at org.jboss.weld.util.reflection.SecureReflectionAccess.run(SecureReflectionAccess.java:52)
        at org.jboss.weld.util.reflection.SecureReflectionAccess.runAsInvocation(SecureReflectionAccess.java:137)
        at org.jboss.weld.util.reflection.SecureReflections.invoke(SecureReflections.java:263)
        at org.jboss.weld.bean.proxy.EnterpriseBeanProxyMethodHandler.invoke(EnterpriseBeanProxyMethodHandler.java:115)
        at org.jboss.weld.bean.proxy.EnterpriseTargetBeanInstance.invoke(EnterpriseTargetBeanInstance.java:56)
        at org.jboss.weld.bean.proxy.ProxyMethodHandler.invoke(ProxyMethodHandler.java:105)
        at ch.puzzle.itc.mobiliar.business.security.boundary.PermissionBoundary$Proxy$_$$_Weld$Proxy$.getAllAvailablePermissions(PermissionBoundary$Proxy$_$$_Weld$Proxy$.java)
        at ch.mobi.itc.mobiliar.rest.permissions.RestrictionsRest.getPermissionEnumValues(RestrictionsRest.java:268)
        at sun.reflect.GeneratedMethodAccessor580.invoke(Unknown Source) [:1.8.0_144]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) [rt.jar:1.8.0_144]
        at java.lang.reflect.Method.invoke(Method.java:498) [rt.jar:1.8.0_144]
        at org.jboss.as.ee.component.ManagedReferenceMethodInterceptor.processInvocation(ManagedReferenceMethodInterceptor.java:52)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.invocation.WeavedInterceptor.processInvocation(WeavedInterceptor.java:53)
        at org.jboss.as.ee.component.interceptors.UserInterceptorFactory$1.processInvocation(UserInterceptorFactory.java:63)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.invocation.InterceptorContext$Invocation.proceed(InterceptorContext.java:375)
        at org.jboss.as.weld.ejb.Jsr299BindingsInterceptor.doMethodInterception(Jsr299BindingsInterceptor.java:86)
        at org.jboss.as.weld.ejb.Jsr299BindingsInterceptor.processInvocation(Jsr299BindingsInterceptor.java:97)
        at org.jboss.as.ee.component.interceptors.UserInterceptorFactory$1.processInvocation(UserInterceptorFactory.java:63)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.invocation.WeavedInterceptor.processInvocation(WeavedInterceptor.java:53)
        at org.jboss.as.ee.component.interceptors.UserInterceptorFactory$1.processInvocation(UserInterceptorFactory.java:63)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.component.invocationmetrics.ExecutionTimeInterceptor.processInvocation(ExecutionTimeInterceptor.java:43)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.jpa.interceptor.SBInvocationInterceptor.processInvocation(SBInvocationInterceptor.java:47)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.weld.ejb.EjbRequestScopeActivationInterceptor.processInvocation(EjbRequestScopeActivationInterceptor.java:73)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.invocation.InitialInterceptor.processInvocation(InitialInterceptor.java:21)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.invocation.ChainedInterceptor.processInvocation(ChainedInterceptor.java:61)
        at org.jboss.as.ee.component.interceptors.ComponentDispatcherInterceptor.processInvocation(ComponentDispatcherInterceptor.java:53)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.component.pool.PooledInstanceInterceptor.processInvocation(PooledInstanceInterceptor.java:51)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.tx.CMTTxInterceptor.invokeInOurTx(CMTTxInterceptor.java:279)
        at org.jboss.as.ejb3.tx.CMTTxInterceptor.required(CMTTxInterceptor.java:331)
        at org.jboss.as.ejb3.tx.CMTTxInterceptor.processInvocation(CMTTxInterceptor.java:243)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.component.interceptors.CurrentInvocationContextInterceptor.processInvocation(CurrentInvocationContextInterceptor.java:41)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.component.invocationmetrics.WaitTimeInterceptor.processInvocation(WaitTimeInterceptor.java:43)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.deployment.processors.StartupAwaitInterceptor.processInvocation(StartupAwaitInterceptor.java:22)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.component.interceptors.ShutDownInterceptorFactory$1.processInvocation(ShutDownInterceptorFactory.java:64)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.component.interceptors.LoggingInterceptor.processInvocation(LoggingInterceptor.java:59)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ee.component.NamespaceContextInterceptor.processInvocation(NamespaceContextInterceptor.java:50)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.component.interceptors.AdditionalSetupInterceptor.processInvocation(AdditionalSetupInterceptor.java:55)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ee.component.TCCLInterceptor.processInvocation(TCCLInterceptor.java:45)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.invocation.ChainedInterceptor.processInvocation(ChainedInterceptor.java:61)
        at org.jboss.as.ee.component.ViewService$View.invoke(ViewService.java:189)
        at org.jboss.as.ee.component.ViewDescription$1.processInvocation(ViewDescription.java:185)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.invocation.ChainedInterceptor.processInvocation(ChainedInterceptor.java:61)
        at org.jboss.as.ee.component.ProxyInvocationHandler.invoke(ProxyInvocationHandler.java:73)
        at ch.mobi.itc.mobiliar.rest.permissions.RestrictionsRest$$$view18.getPermissionEnumValues(Unknown Source)
        at sun.reflect.GeneratedMethodAccessor580.invoke(Unknown Source) [:1.8.0_144]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) [rt.jar:1.8.0_144]
        at java.lang.reflect.Method.invoke(Method.java:498) [rt.jar:1.8.0_144]
        at org.jboss.weld.util.reflection.SecureReflections$13.work(SecureReflections.java:267)
        at org.jboss.weld.util.reflection.SecureReflectionAccess.run(SecureReflectionAccess.java:52)
        at org.jboss.weld.util.reflection.SecureReflectionAccess.runAsInvocation(SecureReflectionAccess.java:137)
        at org.jboss.weld.util.reflection.SecureReflections.invoke(SecureReflections.java:263)
        at org.jboss.weld.bean.proxy.EnterpriseBeanProxyMethodHandler.invoke(EnterpriseBeanProxyMethodHandler.java:115)
        at org.jboss.weld.bean.proxy.EnterpriseTargetBeanInstance.invoke(EnterpriseTargetBeanInstance.java:56)
        at org.jboss.weld.bean.proxy.ProxyMethodHandler.invoke(ProxyMethodHandler.java:105)
        at ch.mobi.itc.mobiliar.rest.permissions.RestrictionsRest$Proxy$_$$_Weld$Proxy$.getPermissionEnumValues(RestrictionsRest$Proxy$_$$_Weld$Proxy$.java)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) [rt.jar:1.8.0_144]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) [rt.jar:1.8.0_144]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) [rt.jar:1.8.0_144]
        at java.lang.reflect.Method.invoke(Method.java:498) [rt.jar:1.8.0_144]
        at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:168)
        at org.jboss.resteasy.core.ResourceMethod.invokeOnTarget(ResourceMethod.java:269)
        at org.jboss.resteasy.core.ResourceMethod.invoke(ResourceMethod.java:227)
        at org.jboss.resteasy.core.ResourceMethod.invoke(ResourceMethod.java:216)
        at org.jboss.resteasy.core.SynchronousDispatcher.getResponse(SynchronousDispatcher.java:583)
        ... 24 more
Caused by: ch.puzzle.itc.mobiliar.common.exception.NotAuthorizedException: Not Authorized!
        at ch.puzzle.itc.mobiliar.business.security.control.PermissionService.throwNotAuthorizedException(PermissionService.java:302)
        at sun.reflect.GeneratedMethodAccessor578.invoke(Unknown Source) [:1.8.0_144]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) [rt.jar:1.8.0_144]
        at java.lang.reflect.Method.invoke(Method.java:498) [rt.jar:1.8.0_144]
        at org.jboss.as.ee.component.ManagedReferenceMethodInterceptor.processInvocation(ManagedReferenceMethodInterceptor.java:52)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.invocation.WeavedInterceptor.processInvocation(WeavedInterceptor.java:53)
        at org.jboss.as.ee.component.interceptors.UserInterceptorFactory$1.processInvocation(UserInterceptorFactory.java:63)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.invocation.InterceptorContext$Invocation.proceed(InterceptorContext.java:375)
        at org.jboss.as.weld.ejb.Jsr299BindingsInterceptor.doMethodInterception(Jsr299BindingsInterceptor.java:86)
        at org.jboss.as.weld.ejb.Jsr299BindingsInterceptor.processInvocation(Jsr299BindingsInterceptor.java:97)
        at org.jboss.as.ee.component.interceptors.UserInterceptorFactory$1.processInvocation(UserInterceptorFactory.java:63)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.invocation.WeavedInterceptor.processInvocation(WeavedInterceptor.java:53)
        at org.jboss.as.ee.component.interceptors.UserInterceptorFactory$1.processInvocation(UserInterceptorFactory.java:63)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.component.invocationmetrics.ExecutionTimeInterceptor.processInvocation(ExecutionTimeInterceptor.java:43)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.jpa.interceptor.SBInvocationInterceptor.processInvocation(SBInvocationInterceptor.java:47)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.weld.ejb.EjbRequestScopeActivationInterceptor.processInvocation(EjbRequestScopeActivationInterceptor.java:73)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.invocation.InitialInterceptor.processInvocation(InitialInterceptor.java:21)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.invocation.ChainedInterceptor.processInvocation(ChainedInterceptor.java:61)
        at org.jboss.as.ee.component.interceptors.ComponentDispatcherInterceptor.processInvocation(ComponentDispatcherInterceptor.java:53)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.component.pool.PooledInstanceInterceptor.processInvocation(PooledInstanceInterceptor.java:51)
        at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288)
        at org.jboss.as.ejb3.tx.CMTTxInterceptor.invokeInCallerTx(CMTTxInterceptor.java:258)
        ... 199 more
```

--- a/AMW_business/src/main/java/ch/puzzle/itc/mobiliar/business/resourcegroup/boundary/ResourceLocator.java
+++ b/AMW_business/src/main/java/ch/puzzle/itc/mobiliar/business/resourcegroup/boundary/ResourceLocator.java
@@ -165,6 +165,13 @@ public class ResourceLocator {
         return resourceRepository.getResourceByNameAndReleaseWithProvidedRelations(name, release);
     }
 
+    public ResourceEntity getResourceByNameAndReleaseWithAllRelations(String name, String releaseName)
+            throws ValidationException {
+        ValidationHelper.validateNotNullOrEmptyChecked(name, releaseName);
+        ReleaseEntity release = releaseLocator.getReleaseByName(releaseName);
+        return resourceRepository.getResourceByNameAndReleaseWithAllRelations(name, release);
+    }
+
 	public List<ResourceEntity> getResourcesByGroupNameWithRelations(String groupName)
 			throws ValidationException {
 		ValidationHelper.validateNotNullOrEmptyChecked(groupName);

--- a/AMW_business/src/main/java/ch/puzzle/itc/mobiliar/business/resourcegroup/control/ResourceRepository.java
+++ b/AMW_business/src/main/java/ch/puzzle/itc/mobiliar/business/resourcegroup/control/ResourceRepository.java
@@ -81,6 +81,14 @@ public class ResourceRepository {
                 .setParameter("name", name.toLowerCase()).setParameter("release", release).getSingleResult();
     }
 
+    public ResourceEntity getResourceByNameAndReleaseWithAllRelations(String name, ReleaseEntity release) {
+        return entityManager
+                .createQuery(
+                        "select r from ResourceEntity r left join fetch r.providedMasterRelations rel left join fetch r.consumedMasterRelations rel left join fetch rel.slaveResource where LOWER(r.name)=:name and r.release=:release",
+                        ResourceEntity.class)
+                .setParameter("name", name.toLowerCase()).setParameter("release", release).getSingleResult();
+    }
+
     public ResourceEntity getResourceByIdWithRelations(int resourceId) {
         String qlString = "SELECT r FROM ResourceEntity r " +
                           "LEFT JOIN FETCH r.consumedMasterRelations rel " +

--- a/AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/RESTApplication.java
+++ b/AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/RESTApplication.java
@@ -24,10 +24,7 @@ import ch.mobi.itc.mobiliar.rest.auditview.AuditViewRest;
 import ch.mobi.itc.mobiliar.rest.deployments.DeploymentDtoCsvBodyWriter;
 import ch.mobi.itc.mobiliar.rest.deployments.DeploymentsRest;
 import ch.mobi.itc.mobiliar.rest.environments.EnvironmentsRest;
-import ch.mobi.itc.mobiliar.rest.exceptions.IllegalArgumentExceptionMapper;
-import ch.mobi.itc.mobiliar.rest.exceptions.IllegalStateExceptionMapper;
-import ch.mobi.itc.mobiliar.rest.exceptions.NoResultExceptionMapper;
-import ch.mobi.itc.mobiliar.rest.exceptions.NotAuthorizedExceptionMapper;
+import ch.mobi.itc.mobiliar.rest.exceptions.*;
 import ch.mobi.itc.mobiliar.rest.health.HealthCheck;
 import ch.mobi.itc.mobiliar.rest.permissions.RestrictionsRest;
 import ch.mobi.itc.mobiliar.rest.resources.*;
@@ -47,12 +44,7 @@ public class RESTApplication extends Application {
         addRestResourceClasses(resources);
         return resources;
     }
-    /**
-     * Do not modify addRestResourceClasses() method.
-     * It is automatically populated with
-     * all resources defined in the project.
-     * If required, comment out calling this method in getClasses().
-     */
+
     private void addRestResourceClasses(Set<Class<?>> resources) {
         resources.add(ReleasesRest.class);
         resources.add(ResourcesRest.class);
@@ -67,10 +59,12 @@ public class RESTApplication extends Application {
         resources.add(EnvironmentsRest.class);
         resources.add(AuditViewRest.class);
         resources.add(RestrictionsRest.class);
-        resources.add(NoResultExceptionMapper.class);
+        resources.add(EJBExceptionMapper.class);
         resources.add(IllegalStateExceptionMapper.class);
-        resources.add(NotAuthorizedExceptionMapper.class);
         resources.add(IllegalArgumentExceptionMapper.class);
+        resources.add(NoResultExceptionMapper.class);
+        resources.add(NotAuthorizedExceptionMapper.class);
+        resources.add(ValidationExceptionMapper.class);
         resources.add(ServerTupleCSVBodyWriter.class);
         resources.add(HealthCheck.class);
     }

--- a/AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/dtos/ResourceRelationDTO.java
+++ b/AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/dtos/ResourceRelationDTO.java
@@ -20,6 +20,7 @@
 
 package ch.mobi.itc.mobiliar.rest.dtos;
 
+import ch.puzzle.itc.mobiliar.business.resourcerelation.entity.AbstractResourceRelationEntity;
 import ch.puzzle.itc.mobiliar.business.resourcerelation.entity.ConsumedResourceRelationEntity;
 import lombok.Data;
 
@@ -33,20 +34,26 @@ import java.util.List;
 @Data
 public class ResourceRelationDTO {
 
+    private final static String CONSUMED = "consumed";
+    private final static String PROVIDED = "provided";
+
     private String relatedResourceName;
+    private String type;
     private String relatedResourceRelease;
 
     private String relationName;
-    private String type;
+    private String relationType;
     private List<TemplateDTO> templates;
 
     ResourceRelationDTO() {
     }
 
-    public ResourceRelationDTO(ConsumedResourceRelationEntity relation) {
+    public ResourceRelationDTO(AbstractResourceRelationEntity relation) {
         relatedResourceName = relation.getSlaveResource().getName();
+        type = relation.getResourceRelationType().getResourceTypeB().getName();
         relatedResourceRelease = relation.getSlaveResource().getRelease().getName();
         relationName = relation.buildIdentifer();
-        type = relation.getResourceRelationType().getResourceTypeB().getName();
+        relationType = relation instanceof ConsumedResourceRelationEntity ? CONSUMED : PROVIDED;
     }
+
 }

--- a/AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/exceptions/NotAuthorizedExceptionMapper.java
+++ b/AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/exceptions/NotAuthorizedExceptionMapper.java
@@ -30,6 +30,6 @@ import javax.ws.rs.ext.Provider;
 public class NotAuthorizedExceptionMapper implements ExceptionMapper<NotAuthorizedException> {
     @Override
     public Response toResponse(NotAuthorizedException exception) {
-        return Response.status(Response.Status.UNAUTHORIZED).build();
+        return Response.status(Response.Status.FORBIDDEN).build();
     }
 }

--- a/AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/exceptions/ValidationExceptionMapper.java
+++ b/AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/exceptions/ValidationExceptionMapper.java
@@ -25,7 +25,6 @@ import javax.ws.rs.ext.ExceptionMapper;
 import javax.ws.rs.ext.Provider;
 
 import ch.puzzle.itc.mobiliar.business.utils.ValidationException;
-import ch.puzzle.itc.mobiliar.common.exception.NotAuthorizedException;
 
 @Provider
 public class ValidationExceptionMapper implements ExceptionMapper<ValidationException> {

--- a/AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/permissions/RestrictionsRest.java
+++ b/AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/permissions/RestrictionsRest.java
@@ -30,7 +30,6 @@ import ch.puzzle.itc.mobiliar.business.security.entity.PermissionEntity;
 import ch.puzzle.itc.mobiliar.business.security.entity.RestrictionEntity;
 import ch.puzzle.itc.mobiliar.business.security.entity.RoleEntity;
 import ch.puzzle.itc.mobiliar.common.exception.AMWException;
-import ch.puzzle.itc.mobiliar.common.exception.NotAuthorizedException;
 import io.swagger.annotations.Api;
 import io.swagger.annotations.ApiOperation;
 import io.swagger.annotations.ApiParam;
@@ -75,8 +74,6 @@ public class RestrictionsRest {
         try {
             id = permissionBoundary.createRestriction(request.getRoleName(), request.getUserName(), request.getPermission().getName(), request.getResourceGroupId(),
                     request.getResourceTypeName(), request.getResourceTypePermission(), request.getContextName(), request.getAction(), delegation);
-        } catch (NotAuthorizedException ne) {
-            return Response.status(FORBIDDEN).build();
         } catch (AMWException e) {
             return Response.status(BAD_REQUEST).entity(new ExceptionDto(e.getMessage())).build();
         }
@@ -104,8 +101,6 @@ public class RestrictionsRest {
         try {
             count = permissionBoundary.createMultipleRestrictions(request.getRoleName(), request.getUserNames(), request.getPermissionNames(), request.getResourceGroupIds(),
                     request.getResourceTypeNames(), request.getResourceTypePermission(), request.getContextNames(), request.getActions(), delegation);
-        } catch (NotAuthorizedException ne) {
-            return Response.status(FORBIDDEN).build();
         } catch (AMWException e) {
             return Response.status(BAD_REQUEST).entity(new ExceptionDto(e.getMessage())).build();
         }
@@ -123,12 +118,7 @@ public class RestrictionsRest {
     // support digit only
     @ApiOperation(value = "Get Restriction by id")
     public Response getRestriction(@ApiParam("Restriction ID") @PathParam("id") Integer id) {
-        final RestrictionEntity restriction;
-        try {
-            restriction = permissionBoundary.findRestriction(id);
-        } catch (NotAuthorizedException ne) {
-            return Response.status(FORBIDDEN).build();
-        }
+        final RestrictionEntity restriction = permissionBoundary.findRestriction(id);
         if (restriction == null) {
             return Response.status(NOT_FOUND).build();
         }
@@ -144,12 +134,7 @@ public class RestrictionsRest {
     @Path("/")
     @ApiOperation(value = "Get all Restrictions")
     public Response getAllRestriction() {
-        final List<RestrictionEntity> restrictions;
-        try {
-            restrictions = permissionBoundary.findAllRestrictions();
-        } catch (NotAuthorizedException ne) {
-            return Response.status(FORBIDDEN).build();
-        }
+        final List<RestrictionEntity> restrictions = permissionBoundary.findAllRestrictions();
         List<RestrictionDTO> restrictionList = new ArrayList<>(restrictions.size());
         for (RestrictionEntity restriction : restrictions) {
             restrictionList.add(new RestrictionDTO(restriction));
@@ -172,8 +157,6 @@ public class RestrictionsRest {
             succcess = permissionBoundary.updateRestriction(id, request.getRoleName(), request.getUserName(), request.getPermission().getName(),
                     request.getResourceGroupId(), request.getResourceTypeName(), request.getResourceTypePermission(),
                     request.getContextName(), request.getAction());
-        } catch (NotAuthorizedException ne) {
-            return Response.status(FORBIDDEN).build();
         } catch (AMWException e) {
             return Response.status(BAD_REQUEST).entity(new ExceptionDto(e.getMessage())).build();
         }
@@ -194,8 +177,6 @@ public class RestrictionsRest {
     public Response deleteRestriction(@ApiParam("Restriction ID") @PathParam("id") Integer id) {
         try {
             permissionBoundary.removeRestriction(id);
-        } catch (NotAuthorizedException ne) {
-            return Response.status(FORBIDDEN).build();
         } catch (AMWException e) {
             return Response.status(NOT_FOUND).entity(new ExceptionDto(e.getMessage())).build();
         }
@@ -211,12 +192,7 @@ public class RestrictionsRest {
     @Path("/roles/")
     @ApiOperation(value = "Get all Roles with restrictions")
     public Response getAllRoles() {
-        final Map<String, List<ch.puzzle.itc.mobiliar.business.security.entity.RestrictionDTO>> allRoles;
-        try {
-            allRoles = permissionBoundary.getAllPermissions();
-        } catch (NotAuthorizedException ne) {
-            return Response.status(FORBIDDEN).build();
-        }
+        final Map<String, List<ch.puzzle.itc.mobiliar.business.security.entity.RestrictionDTO>> allRoles = permissionBoundary.getAllPermissions();
         Map<String, List<RestrictionDTO>> rolesMap = new HashMap<>(allRoles.size());
         // converting business RestrictionDTOs to rest RestrictionDTOs
         for (Map.Entry<String, List<ch.puzzle.itc.mobiliar.business.security.entity.RestrictionDTO>> roleRestrictionList : allRoles.entrySet()) {
@@ -240,12 +216,7 @@ public class RestrictionsRest {
     @Path("/roles/{roleName}")
     @ApiOperation(value = "Get all Restrictions assigned to a specific Role")
     public Response getRoleRestriction(@ApiParam("UserName") @PathParam("roleName") String roleName) {
-        final List<RestrictionEntity> restrictions;
-        try {
-            restrictions = permissionBoundary.getRestrictionsByRoleName(roleName);
-        } catch (NotAuthorizedException ne) {
-            return Response.status(FORBIDDEN).build();
-        }
+        final List<RestrictionEntity> restrictions = permissionBoundary.getRestrictionsByRoleName(roleName);
         List<RestrictionDTO> restrictionList = new ArrayList<>(restrictions.size());
         for (RestrictionEntity restriction : restrictions) {
             restrictionList.add(new RestrictionDTO(restriction));
@@ -262,12 +233,7 @@ public class RestrictionsRest {
     @Path("/roleNames/")
     @ApiOperation(value = "Get all available RoleNames")
     public Response getRoleNames() {
-        final List<RoleEntity> roles;
-        try {
-            roles = permissionBoundary.getAllRoles();
-        } catch (NotAuthorizedException ne) {
-            return Response.status(FORBIDDEN).build();
-        }
+        final List<RoleEntity> roles = permissionBoundary.getAllRoles();
         List<String> roleNameList = new ArrayList<>(roles.size());
         for (RoleEntity role : roles) {
             roleNameList.add(role.getName());
@@ -284,13 +250,7 @@ public class RestrictionsRest {
     @Path("/userRestrictionNames/")
     @ApiOperation(value = "Get all available userRestrictionNames")
     public Response getUserNames() {
-        final List<String> allUserRestrictionNames;
-        try {
-            allUserRestrictionNames = permissionBoundary.getAllUserRestrictionNames();
-        } catch (NotAuthorizedException ne) {
-            return Response.status(FORBIDDEN).build();
-        }
-        return Response.status(OK).entity(allUserRestrictionNames).build();
+        return Response.status(OK).entity(permissionBoundary.getAllUserRestrictionNames()).build();
     }
 
     /**
@@ -302,12 +262,7 @@ public class RestrictionsRest {
     @Path("/permissionEnumValues/")
     @ApiOperation(value = "Get all available PermissionEnum values")
     public Response getPermissionEnumValues() {
-        final List<PermissionEntity> permissions;
-        try {
-            permissions = permissionBoundary.getAllAvailablePermissions();
-        } catch (NotAuthorizedException ne) {
-            return Response.status(FORBIDDEN).build();
-        }
+        final List<PermissionEntity> permissions = permissionBoundary.getAllAvailablePermissions();
         List<PermissionDTO> permissionNameList = new ArrayList<>(permissions.size());
         for (PermissionEntity permission : permissions) {
             permissionNameList.add(new PermissionDTO(Permission.valueOf(permission.getValue())));
@@ -324,12 +279,7 @@ public class RestrictionsRest {
     @Path("/users/")
     @ApiOperation(value = "Get all UserRestriction with their Restrictions")
     public Response getAllUsers() {
-        final List<RestrictionEntity> allUserRestrictions;
-        try {
-            allUserRestrictions = permissionBoundary.getAllUserRestriction();
-        } catch (NotAuthorizedException ne) {
-            return Response.status(FORBIDDEN).build();
-        }
+        final List<RestrictionEntity> allUserRestrictions = permissionBoundary.getAllUserRestriction();
         Map<String, List<RestrictionDTO>> usersMap = new HashMap<>(allUserRestrictions.size());
         for (RestrictionEntity restriction : allUserRestrictions) {
             String userName = restriction.getUser().getName();
@@ -351,12 +301,7 @@ public class RestrictionsRest {
     @Path("/users/{userName}")
     @ApiOperation(value = "Get all Restrictions assigned to a specific UserRestriction")
     public Response getUserRestriction(@ApiParam("UserName") @PathParam("userName") String userName) {
-        final List<RestrictionEntity> restrictions;
-        try {
-            restrictions = permissionBoundary.getRestrictionsByUserName(userName);
-        } catch (NotAuthorizedException ne) {
-            return Response.status(FORBIDDEN).build();
-        }
+        final List<RestrictionEntity> restrictions = permissionBoundary.getRestrictionsByUserName(userName);
         List<RestrictionDTO> restrictionList = new ArrayList<>(restrictions.size());
         for (RestrictionEntity restriction : restrictions) {
             restrictionList.add(new RestrictionDTO(restriction));
@@ -373,12 +318,7 @@ public class RestrictionsRest {
     @Path("/ownRestrictions/")
     @ApiOperation(value = "Get all Restrictions assigned to a specific UserRestriction")
     public Response getCallerRestrictions() {
-        final List<RestrictionEntity> restrictions;
-        try {
-            restrictions = permissionBoundary.getAllCallerRestrictions();
-        } catch (NotAuthorizedException ne) {
-            return Response.status(FORBIDDEN).build();
-        }
+        final List<RestrictionEntity> restrictions = permissionBoundary.getAllCallerRestrictions();
         List<RestrictionDTO> restrictionList = new ArrayList<>(restrictions.size());
         for (RestrictionEntity restriction : restrictions) {
             restrictionList.add(new RestrictionDTO(restriction));

--- a/AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/permissions/RestrictionsRest.java
+++ b/AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/permissions/RestrictionsRest.java
@@ -24,13 +24,13 @@ import ch.mobi.itc.mobiliar.rest.dtos.PermissionDTO;
 import ch.mobi.itc.mobiliar.rest.dtos.RestrictionDTO;
 import ch.mobi.itc.mobiliar.rest.dtos.RestrictionsCreationDTO;
 import ch.mobi.itc.mobiliar.rest.exceptions.ExceptionDto;
-import ch.puzzle.itc.mobiliar.business.environment.boundary.ContextLocator;
 import ch.puzzle.itc.mobiliar.business.security.boundary.PermissionBoundary;
 import ch.puzzle.itc.mobiliar.business.security.entity.Permission;
 import ch.puzzle.itc.mobiliar.business.security.entity.PermissionEntity;
 import ch.puzzle.itc.mobiliar.business.security.entity.RestrictionEntity;
 import ch.puzzle.itc.mobiliar.business.security.entity.RoleEntity;
 import ch.puzzle.itc.mobiliar.common.exception.AMWException;
+import ch.puzzle.itc.mobiliar.common.exception.NotAuthorizedException;
 import io.swagger.annotations.Api;
 import io.swagger.annotations.ApiOperation;
 import io.swagger.annotations.ApiParam;
@@ -54,8 +54,6 @@ public class RestrictionsRest {
 
     @Inject
     PermissionBoundary permissionBoundary;
-    @Inject
-    ContextLocator contextLocator;
 
     /**
      * Creates a new restriction and returns the newly created restriction.
@@ -77,6 +75,8 @@ public class RestrictionsRest {
         try {
             id = permissionBoundary.createRestriction(request.getRoleName(), request.getUserName(), request.getPermission().getName(), request.getResourceGroupId(),
                     request.getResourceTypeName(), request.getResourceTypePermission(), request.getContextName(), request.getAction(), delegation);
+        } catch (NotAuthorizedException ne) {
+            return Response.status(FORBIDDEN).build();
         } catch (AMWException e) {
             return Response.status(BAD_REQUEST).entity(new ExceptionDto(e.getMessage())).build();
         }
@@ -104,6 +104,8 @@ public class RestrictionsRest {
         try {
             count = permissionBoundary.createMultipleRestrictions(request.getRoleName(), request.getUserNames(), request.getPermissionNames(), request.getResourceGroupIds(),
                     request.getResourceTypeNames(), request.getResourceTypePermission(), request.getContextNames(), request.getActions(), delegation);
+        } catch (NotAuthorizedException ne) {
+            return Response.status(FORBIDDEN).build();
         } catch (AMWException e) {
             return Response.status(BAD_REQUEST).entity(new ExceptionDto(e.getMessage())).build();
         }
@@ -121,7 +123,12 @@ public class RestrictionsRest {
     // support digit only
     @ApiOperation(value = "Get Restriction by id")
     public Response getRestriction(@ApiParam("Restriction ID") @PathParam("id") Integer id) {
-        RestrictionEntity restriction = permissionBoundary.findRestriction(id);
+        final RestrictionEntity restriction;
+        try {
+            restriction = permissionBoundary.findRestriction(id);
+        } catch (NotAuthorizedException ne) {
+            return Response.status(FORBIDDEN).build();
+        }
         if (restriction == null) {
             return Response.status(NOT_FOUND).build();
         }
@@ -137,11 +144,17 @@ public class RestrictionsRest {
     @Path("/")
     @ApiOperation(value = "Get all Restrictions")
     public Response getAllRestriction() {
-        List<RestrictionDTO> restrictions = new ArrayList<>();
-        for (RestrictionEntity restrictionEntity : permissionBoundary.findAllRestrictions()) {
-            restrictions.add(new RestrictionDTO(restrictionEntity));
+        final List<RestrictionEntity> restrictions;
+        try {
+            restrictions = permissionBoundary.findAllRestrictions();
+        } catch (NotAuthorizedException ne) {
+            return Response.status(FORBIDDEN).build();
         }
-        return Response.status(OK).entity(restrictions).build();
+        List<RestrictionDTO> restrictionList = new ArrayList<>(restrictions.size());
+        for (RestrictionEntity restriction : restrictions) {
+            restrictionList.add(new RestrictionDTO(restriction));
+        }
+        return Response.status(OK).entity(restrictionList).build();
     }
 
     /**
@@ -159,6 +172,8 @@ public class RestrictionsRest {
             succcess = permissionBoundary.updateRestriction(id, request.getRoleName(), request.getUserName(), request.getPermission().getName(),
                     request.getResourceGroupId(), request.getResourceTypeName(), request.getResourceTypePermission(),
                     request.getContextName(), request.getAction());
+        } catch (NotAuthorizedException ne) {
+            return Response.status(FORBIDDEN).build();
         } catch (AMWException e) {
             return Response.status(BAD_REQUEST).entity(new ExceptionDto(e.getMessage())).build();
         }
@@ -179,6 +194,8 @@ public class RestrictionsRest {
     public Response deleteRestriction(@ApiParam("Restriction ID") @PathParam("id") Integer id) {
         try {
             permissionBoundary.removeRestriction(id);
+        } catch (NotAuthorizedException ne) {
+            return Response.status(FORBIDDEN).build();
         } catch (AMWException e) {
             return Response.status(NOT_FOUND).entity(new ExceptionDto(e.getMessage())).build();
         }
@@ -194,8 +211,13 @@ public class RestrictionsRest {
     @Path("/roles/")
     @ApiOperation(value = "Get all Roles with restrictions")
     public Response getAllRoles() {
-        Map<String, List<RestrictionDTO>> rolesMap = new HashMap<>();
-        final Map<String, List<ch.puzzle.itc.mobiliar.business.security.entity.RestrictionDTO>> allRoles = permissionBoundary.getAllPermissions();
+        final Map<String, List<ch.puzzle.itc.mobiliar.business.security.entity.RestrictionDTO>> allRoles;
+        try {
+            allRoles = permissionBoundary.getAllPermissions();
+        } catch (NotAuthorizedException ne) {
+            return Response.status(FORBIDDEN).build();
+        }
+        Map<String, List<RestrictionDTO>> rolesMap = new HashMap<>(allRoles.size());
         // converting business RestrictionDTOs to rest RestrictionDTOs
         for (Map.Entry<String, List<ch.puzzle.itc.mobiliar.business.security.entity.RestrictionDTO>> roleRestrictionList : allRoles.entrySet()) {
             String roleName = roleRestrictionList.getKey();
@@ -218,8 +240,13 @@ public class RestrictionsRest {
     @Path("/roles/{roleName}")
     @ApiOperation(value = "Get all Restrictions assigned to a specific Role")
     public Response getRoleRestriction(@ApiParam("UserName") @PathParam("roleName") String roleName) {
-        List<RestrictionDTO> restrictionList = new ArrayList<>();
-        final List<RestrictionEntity> restrictions = permissionBoundary.getRestrictionsByRoleName(roleName);
+        final List<RestrictionEntity> restrictions;
+        try {
+            restrictions = permissionBoundary.getRestrictionsByRoleName(roleName);
+        } catch (NotAuthorizedException ne) {
+            return Response.status(FORBIDDEN).build();
+        }
+        List<RestrictionDTO> restrictionList = new ArrayList<>(restrictions.size());
         for (RestrictionEntity restriction : restrictions) {
             restrictionList.add(new RestrictionDTO(restriction));
         }
@@ -235,8 +262,13 @@ public class RestrictionsRest {
     @Path("/roleNames/")
     @ApiOperation(value = "Get all available RoleNames")
     public Response getRoleNames() {
-        List<String> roleNameList = new ArrayList<>();
-        final List<RoleEntity> roles = permissionBoundary.getAllRoles();
+        final List<RoleEntity> roles;
+        try {
+            roles = permissionBoundary.getAllRoles();
+        } catch (NotAuthorizedException ne) {
+            return Response.status(FORBIDDEN).build();
+        }
+        List<String> roleNameList = new ArrayList<>(roles.size());
         for (RoleEntity role : roles) {
             roleNameList.add(role.getName());
         }
@@ -252,7 +284,13 @@ public class RestrictionsRest {
     @Path("/userRestrictionNames/")
     @ApiOperation(value = "Get all available userRestrictionNames")
     public Response getUserNames() {
-        return Response.status(OK).entity(permissionBoundary.getAllUserRestrictionNames()).build();
+        final List<String> allUserRestrictionNames;
+        try {
+            allUserRestrictionNames = permissionBoundary.getAllUserRestrictionNames();
+        } catch (NotAuthorizedException ne) {
+            return Response.status(FORBIDDEN).build();
+        }
+        return Response.status(OK).entity(allUserRestrictionNames).build();
     }
 
     /**
@@ -264,8 +302,13 @@ public class RestrictionsRest {
     @Path("/permissionEnumValues/")
     @ApiOperation(value = "Get all available PermissionEnum values")
     public Response getPermissionEnumValues() {
-        List<PermissionDTO> permissionNameList = new ArrayList<>();
-        final List<PermissionEntity> permissions = permissionBoundary.getAllAvailablePermissions();
+        final List<PermissionEntity> permissions;
+        try {
+            permissions = permissionBoundary.getAllAvailablePermissions();
+        } catch (NotAuthorizedException ne) {
+            return Response.status(FORBIDDEN).build();
+        }
+        List<PermissionDTO> permissionNameList = new ArrayList<>(permissions.size());
         for (PermissionEntity permission : permissions) {
             permissionNameList.add(new PermissionDTO(Permission.valueOf(permission.getValue())));
         }
@@ -281,8 +324,13 @@ public class RestrictionsRest {
     @Path("/users/")
     @ApiOperation(value = "Get all UserRestriction with their Restrictions")
     public Response getAllUsers() {
-        Map<String, List<RestrictionDTO>> usersMap = new HashMap<>();
-        final List<RestrictionEntity> allUserRestrictions = permissionBoundary.getAllUserRestriction();
+        final List<RestrictionEntity> allUserRestrictions;
+        try {
+            allUserRestrictions = permissionBoundary.getAllUserRestriction();
+        } catch (NotAuthorizedException ne) {
+            return Response.status(FORBIDDEN).build();
+        }
+        Map<String, List<RestrictionDTO>> usersMap = new HashMap<>(allUserRestrictions.size());
         for (RestrictionEntity restriction : allUserRestrictions) {
             String userName = restriction.getUser().getName();
             if (!usersMap.containsKey(userName)) {
@@ -303,8 +351,14 @@ public class RestrictionsRest {
     @Path("/users/{userName}")
     @ApiOperation(value = "Get all Restrictions assigned to a specific UserRestriction")
     public Response getUserRestriction(@ApiParam("UserName") @PathParam("userName") String userName) {
-        List<RestrictionDTO> restrictionList = new ArrayList<>();
-        for (RestrictionEntity restriction : permissionBoundary.getRestrictionsByUserName(userName)) {
+        final List<RestrictionEntity> restrictions;
+        try {
+            restrictions = permissionBoundary.getRestrictionsByUserName(userName);
+        } catch (NotAuthorizedException ne) {
+            return Response.status(FORBIDDEN).build();
+        }
+        List<RestrictionDTO> restrictionList = new ArrayList<>(restrictions.size());
+        for (RestrictionEntity restriction : restrictions) {
             restrictionList.add(new RestrictionDTO(restriction));
         }
         return Response.status(OK).entity(restrictionList).build();
@@ -319,8 +373,14 @@ public class RestrictionsRest {
     @Path("/ownRestrictions/")
     @ApiOperation(value = "Get all Restrictions assigned to a specific UserRestriction")
     public Response getCallerRestrictions() {
-        List<RestrictionDTO> restrictionList = new ArrayList<>();
-        for (RestrictionEntity restriction : permissionBoundary.getAllCallerRestrictions()) {
+        final List<RestrictionEntity> restrictions;
+        try {
+            restrictions = permissionBoundary.getAllCallerRestrictions();
+        } catch (NotAuthorizedException ne) {
+            return Response.status(FORBIDDEN).build();
+        }
+        List<RestrictionDTO> restrictionList = new ArrayList<>(restrictions.size());
+        for (RestrictionEntity restriction : restrictions) {
             restrictionList.add(new RestrictionDTO(restriction));
         }
         return Response.status(OK).entity(restrictionList).build();

--- a/AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/resources/ResourceRelationsRest.java
+++ b/AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/resources/ResourceRelationsRest.java
@@ -31,6 +31,7 @@ import ch.puzzle.itc.mobiliar.business.resourcerelation.boundary.RelationEditor;
 import ch.puzzle.itc.mobiliar.business.resourcerelation.boundary.ResourceRelationLocator;
 import ch.puzzle.itc.mobiliar.business.resourcerelation.entity.AbstractResourceRelationEntity;
 import ch.puzzle.itc.mobiliar.business.resourcerelation.entity.ConsumedResourceRelationEntity;
+import ch.puzzle.itc.mobiliar.business.resourcerelation.entity.ProvidedResourceRelationEntity;
 import ch.puzzle.itc.mobiliar.business.utils.ValidationException;
 import ch.puzzle.itc.mobiliar.common.exception.ElementAlreadyExistsException;
 import ch.puzzle.itc.mobiliar.common.exception.ResourceNotFoundException;
@@ -73,25 +74,38 @@ public class ResourceRelationsRest {
     ResourceRelationTemplatesRest resourceRelationTemplatesRest;
 
     @GET
-    @ApiOperation(value = "Get all relations of the current resource")
+    @ApiOperation(value = "Get all relations of a resource in a specific release, optionally filtered by a resource type")
     public List<ResourceRelationDTO> getResourceRelations() throws ValidationException {
         return getResourceRelations(resourceGroupName, releaseName, resourceType);
     }
 
     List<ResourceRelationDTO> getResourceRelations(String resourceGroupName, String releaseName, String resourceType) throws ValidationException {
-        ResourceEntity resource = resourceLocator.getResourceByNameAndReleaseWithConsumedRelations(resourceGroupName, releaseName);
+        ResourceEntity resource = resourceLocator.getResourceByNameAndReleaseWithAllRelations(resourceGroupName, releaseName);
         List<ResourceRelationDTO> resourceRelations = new ArrayList<>();
         for (ConsumedResourceRelationEntity relation : resource.getConsumedMasterRelations()) {
-            if (resourceType != null && !relation.getResourceRelationType().getResourceTypeB().getName().equals(resourceType)) {
-                continue;
+            ResourceRelationDTO resRel = createResourceRelationDTO(resourceGroupName, releaseName, resourceType, relation);
+            if (resRel != null) {
+                resourceRelations.add(resRel);
             }
-            ResourceRelationDTO resRel = new ResourceRelationDTO(relation);
-            List<TemplateDTO> templates = resourceRelationTemplatesRest.getResourceRelationTemplates(resourceGroupName, releaseName,
-                    relation.getSlaveResource().getName(), relation.getSlaveResource().getRelease().getName(), "");
-            addTemplates(resRel, templates);
-            resourceRelations.add(resRel);
+        }
+        for (ProvidedResourceRelationEntity relation : resource.getProvidedMasterRelations()) {
+            ResourceRelationDTO resRel = createResourceRelationDTO(resourceGroupName, releaseName, resourceType, relation);
+            if (resRel != null) {
+                resourceRelations.add(resRel);
+            }
         }
         return resourceRelations;
+    }
+
+    private ResourceRelationDTO createResourceRelationDTO(String resourceGroupName, String releaseName, String resourceType, AbstractResourceRelationEntity relation) throws ValidationException {
+        if (resourceType != null && !relation.getResourceRelationType().getResourceTypeB().getName().equals(resourceType)) {
+            return null;
+        }
+        ResourceRelationDTO resRel = new ResourceRelationDTO(relation);
+        List<TemplateDTO> templates = resourceRelationTemplatesRest.getResourceRelationTemplates(resourceGroupName, releaseName,
+                relation.getSlaveResource().getName(), relation.getSlaveResource().getRelease().getName(), "");
+        addTemplates(resRel, templates);
+        return resRel;
     }
 
     @Path("/{relatedResourceGroupName}")

